### PR TITLE
Error for use super in null extension

### LINKED ISSUE #3696: Error when extending null
```ts
    class C extends null {
        constructor() {
            return Object.create(null);
        }
    }
```
generates error:
> error TS2377: Constructors for derived classes must contain a 'super' call.

in this case, a call to `super` is guaranteed to fail.

### PR DESCRIPTION
**Pull request by yuit**
Fix  #3696.

### CODE REVIEW

File: tests/baselines/reference/classExtendsNull.errors.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    class C extends null {
+        constructor() {
+        ~~~~~~~~~~~~~~~
+            super();
+    ~~~~~~~~~~~~~~~~
+            return Object.create(null);
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        }
+    ~~~~~
+!!! error TS17005: A constructor can not contain super call when a class extends null
```
Comment (yuit) [inline tests/baselines/reference/classExtendsNull.errors.txt]: Fix spelling

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -2445,5 +2445,9 @@
     "Cannot use JSX unless the '--jsx' flag is provided.": {
         "category": "Error",
         "code": 17004
+    },
+    "A constructor can not contain super call when a class extends null": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: A constructor cannot contain a 'super' call when its containing class extends 'null'.

File: src/compiler/checker.ts [hunk 3]
```diff
@@ -10248,19 +10248,27 @@ namespace ts {
...
                     // - The containing class is a derived class.
                     // - The constructor declares parameter properties
                     //   or the containing class declares instance member variables with initializers.
+                    let statements = (<Block>node.body).statements;
                     let superCallShouldBeFirst =
                         forEach((<ClassDeclaration>node.parent).members, isInstancePropertyWithInitializer) ||
                         forEach(node.parameters, p => p.flags & (NodeFlags.Public | NodeFlags.Private | NodeFlags.Protected));
 
                     if (superCallShouldBeFirst) {
-                        let statements = (<Block>node.body).statements;
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: Nit: I'd say move this back.

**Comment by DanielRosenwasser** [comment]
:+1:

File: src/compiler/diagnosticMessages.json [hunk 4]
```diff
@@ -2445,5 +2445,9 @@
     "Cannot use JSX unless the '--jsx' flag is provided.": {
         "category": "Error",
         "code": 17004
+    },
+    "A constructor cannot contain 'super' call when a class extends 'null'": {
```
Comment (JsonFreeman) [inline src/compiler/diagnosticMessages.json]: "a 'super' call" and "when its class extends 'null'"

File: src/compiler/checker.ts [hunk 5]
```diff
@@ -10248,19 +10248,27 @@ namespace ts {
             // TS 1.0 spec (April 2014): 8.3.2
             // Constructors of classes with no extends clause may not contain super calls, whereas
             // constructors of derived classes must contain at least one super call somewhere in their function body.
-            if (getClassExtendsHeritageClauseElement(<ClassDeclaration>node.parent)) {
+            let containingClassDecl = <ClassDeclaration>node.parent;
+            if (getClassExtendsHeritageClauseElement(containingClassDecl)) {
+                let symbol = getSymbolOfNode(containingClassDecl);
```
Comment (JsonFreeman) [inline src/compiler/checker.ts]: containingClassSymbol

File: src/compiler/checker.ts [hunk 6]
```diff
@@ -10248,19 +10248,27 @@ namespace ts {
             // TS 1.0 spec (April 2014): 8.3.2
             // Constructors of classes with no extends clause may not contain super calls, whereas
             // constructors of derived classes must contain at least one super call somewhere in their function body.
-            if (getClassExtendsHeritageClauseElement(<ClassDeclaration>node.parent)) {
+            let containingClassDecl = <ClassDeclaration>node.parent;
+            if (getClassExtendsHeritageClauseElement(containingClassDecl)) {
+                let symbol = getSymbolOfNode(containingClassDecl);
+                let type = <InterfaceType>getDeclaredTypeOfSymbol(symbol);
```
Comment (JsonFreeman) [inline src/compiler/checker.ts]: containingClassInstanceType

Comment (JsonFreeman) [inline src/compiler/checker.ts]: (if you think it makes it clearer)

**Comment by JsonFreeman** [comment]
:+1:

**Pull request merged**